h/streamer/messages.py

```diff
@@ -140,8 +147,8 @@ def _generate_annotation_event(message, socket, annotation, user_nipsad):
     base_url = socket.registry.settings.get('h.app_url',
                                             'http://localhost:5000')
     links_service = LinksService(base_url, socket.registry)
-    serialized = presenters.AnnotationJSONPresenter(annotation,
-                                                    links_service).asdict()
+    resource = AnnotationResource(annotation, group_service)
```

Comment:
As discussed in [Slack](https://hypothes-is.slack.com/archives/backend/p1484318794001939), this does not cache groups in-between annotation events, but when generating the responses for each individual clients for one annotation events.